packages/block-library/src/buttons/edit.js

```diff
@@ -20,16 +28,42 @@ const alignmentHooksSetting = {
 	isEmbedButton: true,
 };
 
-function ButtonsEdit() {
+function ButtonsEdit( { attributes, setAttributes } ) {
+	const { orientation } = attributes;
+
 	return (
-		<Block.div>
+		<Block.div
+			className={ classnames( {
+				'is-vertical': orientation === 'vertical',
+			} ) }
+		>
 			<AlignmentHookSettingsProvider value={ alignmentHooksSetting }>
 				<InnerBlocks
 					allowedBlocks={ ALLOWED_BLOCKS }
 					template={ BUTTONS_TEMPLATE }
-					__experimentalMoverDirection="horizontal"
+					__experimentalMoverDirection={ orientation }
 				/>
 			</AlignmentHookSettingsProvider>
+			<InspectorControls>
+				<PanelBody title={ __( 'Buttons Settings' ) }>
+					<RadioControl
+						label={ __( 'Orientation' ) }
+						selected={ orientation }
+						options={ [
+							{ label: __( 'Horizontal' ), value: 'horizontal' },
+							{
+								label: __( 'Vertical' ),
+								value: 'vertical',
+							},
+						] }
```

Comment:
To avoid recreating this array on every render, we could extract this to a file-level constant (probably called `ORIENTATION_OPTIONS`), similar to the existing `ALLOWED_BLOCKS` and `BUTTONS_TEMPLATE` constants.

Reply:
Should we maybe consider this on a larger case scenario? Probably consider other blocks which have orientation options as well just for the sake of consistency? @talldan what do you think?

Reply:
@ZebulanStanphill suggestion seems good to me. 👍 

I don't think it's worth worrying about other blocks at the moment.

Reply:
Thank you for your feedback @ZebulanStanphill and @talldan ! I've implemented the suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/20160/commits/a874f75ae85cf8f53b626fc738ad6c11243e199c). Does it look good?

Reply:
Looks good to me.